main.go

```diff
@@ -311,3 +321,23 @@ func run(ctrlConfig spirev1alpha1.ControllerManagerConfig, options ctrl.Options)
 
 	return nil
 }
+
+func autoDetectClusterDomain() string {
+	cname, err := net.LookupCNAME(k8sDefaultService)
+	if err != nil {
+		setupLog.Error(err, "unable to autodetect cluster domain")
+		return ""
+	}
+
+	re := regexp.MustCompile(regexp.QuoteMeta(k8sDefaultService) + `\.(.*)\.`)
```

Comment:
We should ensure this isn't a substring match by enforcing a whole line match with ^ and $. Further the subgroup match between the dots should be non-empty. Also, the last dot is optional (LookupCName will accept and return it either way):

```suggestion
	re := regexp.MustCompile("^" + regexp.QuoteMeta(k8sDefaultService) + `\.(.+)\.?$`)
```

Reply:
N/A anymore since regex removed. Good improvements though.